src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -190,86 +215,148 @@ cbuf_comp_info_bin_get(struct cbuf_comp_info *cci, int sz)
 }
 
 static struct cbuf_bin *
-cbuf_comp_info_bin_add(struct cbuf_comp_info *cci, int sz)
+cbuf_comp_info_bin_add(struct cbuf_comp_info *cci, unsigned long sz)
 {
-	if (sz == CBUF_MAX_NSZ) return NULL;
 	cci->cbufs[cci->nbin].size = sz;
 	cci->nbin++;
 
 	return &cci->cbufs[cci->nbin-1];
 }
 
 static int
-cbuf_map(spdid_t spdid, vaddr_t daddr, void *page, int size, int flags)
+cbuf_map(spdid_t spdid, vaddr_t daddr, void *page, unsigned long size, int flags)
 {
-	int off;
-	assert(size == (int)round_to_page(size));
+	unsigned long off;
+	assert(size == round_to_page(size));
 	assert(daddr);
 	assert(page);
 	for (off = 0 ; off < size ; off += PAGE_SIZE) {
 		vaddr_t d = daddr + off;
-		if (d != (mman_alias_page(cos_spd_id(), ((vaddr_t)page) + off,
-						spdid, d, flags))) {
-			printd("couldn't alias(%d, %x, %d, %x, %d)\n", cos_spd_id(), ((vaddr_t)page) + off, spdid, d, flags);
-			assert(0); /* TODO: roll back the aliases, etc... */
+		if (unlikely(d != (mman_alias_page(cos_spd_id(), ((vaddr_t)page) + off,
+						   spdid, d, flags)))) {
+			for(d = daddr+off-PAGE_SIZE; d >= daddr; d -= PAGE_SIZE) {
+				mman_revoke_page(spdid, d, 0);
+			}
+			return -ENOMEM;
 		}
 	}
 	return 0;
 }
 
+/* map the memory from address p and size sz to the component spdid with 
+ * permission flags. if p is NULL allocate a piece of new memory
+ * return spdid's address to daddr, manager's virtual address to page*/
 static int
-cbuf_alloc_map(spdid_t spdid, vaddr_t *daddr, void **page, int size)
+cbuf_alloc_map(spdid_t spdid, vaddr_t *daddr, void **page, void *p, unsigned long sz, int flags)
```

Comment:
It looks like there are no uses where p is actually passed. If it is unused I wouldn't keep it. The function is supposed to alloc and map. If you don't need to alloc, you could just map...